File: AL/nadongbin/01-greedy/adventurerGuild/adventurerGuild.ts
function adventurerGuild(adventurers: number[]) {
  adventurers.sort((a, b) => a - b)

  let result = 0
  let count = 0

  for (let i = 0; i < adventurers.length; i++) {
    count++
    if (count >= adventurers[i]) {
      result++
      count = 0
    }
  }

  return result
}

console.log(adventurerGuild([2, 1, 2, 3, 2]))

File: AL/nadongbin/01-greedy/adventurerGuild/adventurerGuild.go
package main

import (
	"fmt"
	"sort"
)

func adventurerGuild(adventurers []int) int {
	sort.Ints(adventurers)

	count := 0
	guild := 0
	for _, adventurer := range adventurers {
		count++
		if count >= adventurer {
			guild++
			count = 0
		}
	}

	return guild
}

func main() {
	fmt.Println(adventurerGuild([]int{2, 3, 1, 2, 2}))
}
